.github/workflows/create_destroy_test_vm.yaml

```
@@ -0,0 +1,257 @@
+          sleep 30s
+          NEW_DROPLET_IP=$(doctl compute droplet get $NEW_DROPLET_NAME \
+            --format PublicIPv4 --no-header)
+          echo "NEW_DROPLET_IP=$NEW_DROPLET_IP" >> $GITHUB_ENV
+
+          # we only need records for frontend, media server and api server
+          doctl compute domain records create unstructured.studio --record-type A --record-name \
+            $FRONTEND_DOMAIN --record-data $NEW_DROPLET_IP --record-ttl 60
+          doctl compute domain records create unstructured.studio --record-type A --record-name \
+            $API_DOMAIN --record-data $NEW_DROPLET_IP --record-ttl 60
+          doctl compute domain records create unstructured.studio --record-type A --record-name \
+            $MEDIA_DOMAIN --record-data $NEW_DROPLET_IP --record-ttl 60
+
+      - name: Copy file via scp
+        uses: appleboy/scp-action@master
+        with:
+          host: ${{env.NEW_DROPLET_IP}}
+          username: ${{ secrets.DO_BACKEND_USERNAME }}
+          key: ${{ secrets.DO_SSHKEY }}
+          source: "."
+          target: "/home/zubhub"
+#################################################################################
+
+#################################################################################
+# The following steps are for building docker images and pushing them to dockerhub
+##################################################################################
+      - name: create a .env file in the zubhub_frontend/zubhub/ directory
+        run: |
+
+          # create env file for frontend.
+          # This vm is only for testing so we don't need to worry about secrets being exposed
+          # (provided we don't put sensitive info in the env file)
+          cat << EOF > ./zubhub_frontend/zubhub/.env
+          GENERATE_SOURCEMAP=false
+          DISABLE_ESLINT_PLUGIN=true
+          REACT_APP_NODE_ENV=development
+          REACT_APP_BACKEND_DEVELOPMENT_URL=https://${{env.API_DOMAIN}}.unstructured.studio
+          REACT_APP_BACKEND_PRODUCTION_URL=https://${{env.API_DOMAIN}}.unstructured.studio
+          REACT_APP_DOSPACE_ACCESS_KEY_ID=
+          REACT_APP_DOSPACE_ACCESS_SECRET_KEY=
+          REACT_APP_VIDEO_UPLOAD_URL=
+          REACT_APP_VIDEO_FOLDER_NAME=videos
+          REACT_APP_DEV_VIDEO_FOLDER_NAME=dev_videos
+          REACT_APP_VIDEO_UPLOAD_PRESET_NAME=video_upload_preset
+          REACT_APP_DEV_VIDEO_UPLOAD_PRESET_NAME=dev_video_upload_preset
+          EOF
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push frontend
+        id: docker_build_frontend
+        uses: docker/build-push-action@v2
```

The version 5 has similar implementation and more optimized in terms of resources usage.

```suggestion
        uses: docker/build-push-action@v5
```